File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/AttributeSetInstanceStateEventId.java
package org.dddml.wms.domain;


public class AttributeSetInstanceStateEventId
{
    private String attributeSetInstanceId;

    public String getAttributeSetInstanceId()
    {
        return this.attributeSetInstanceId;
    }

    public void setAttributeSetInstanceId(String attributeSetInstanceId)
    {
        this.attributeSetInstanceId = attributeSetInstanceId;
    }

    private Long version;

    public Long getVersion()
    {
        return this.version;
    }

    public void setVersion(Long version)
    {
        this.version = version;
    }

    AttributeSetInstanceStateEventId ()
    {
    }

    public AttributeSetInstanceStateEventId (String attributeSetInstanceId, Long version)
    {
        this.attributeSetInstanceId = attributeSetInstanceId;
        this.version = version;
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/AttributeSetInstanceStateEventId.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\StringIdInterface;

class AttributeSetInstanceStateEventId implements StringIdInterface
{
    /**
     * @Type("string")
     */
    private $attributeSetInstanceId;

    /**
     * @return string
     */
    public function getAttributeSetInstanceId()
    {
        return $this->attributeSetInstanceId;
    }

    /**
     * @param string $attributeSetInstanceId
     */
    public function setAttributeSetInstanceId($attributeSetInstanceId)
    {
        $this->attributeSetInstanceId = $attributeSetInstanceId;
    }

    /**
     * @Type("Dddml\Serializer\Type\Long")
     */
    private $version;

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->version;
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->version = $version;
    }



    /**
     * @var AttributeSetInstanceStateEventIdFlattenedDto
     */
    private $idFlattenedDto;

    /**
     * @return string
     */
    public function toString()
    {
        if (!$this->idFlattenedDto) {
            $this->idFlattenedDto = new AttributeSetInstanceStateEventIdFlattenedDto($this);
        }

        return $this->idFlattenedDto->toString();
    }

    /**
     * @param string $idStr
     *
     * @return AttributeSetInstanceStateEventId
     */
    public static function createFromString($idStr)
    {
        return (new AttributeSetInstanceStateEventIdFlattenedDto())
            ->fromString($idStr)
            ->toAttributeSetInstanceStateEventId();
    }


}

